### PR TITLE
Add explicit authorization interface

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**
Provide a common interface, to support extending authorize in future
And also prevent the identities from being persisted, so saves a little memory

### CODE REVIEW

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -321,6 +315,10 @@ func (s *DiscoveryServer) Stream(stream DiscoveryStream) error {
 	// reqChannel and the connection not being enqueued for pushes to pushChannel until the
 	// initialization is complete.
 	<-con.initialized
+	// authorize client
```
Comment (ramaraochavali) [inline pilot/pkg/xds/ads.go:318]: Should n't this be called before we start go routine for receive? May be move authenticate and authorize to initConnection?

Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go:318]: no, the receive go routine initialize proxy, authorize depend on it.

Comment (ramaraochavali) [reply pilot/pkg/xds/ads.go:318]: Ah. Got it

**Review by ramaraochavali — APPROVED** [review]

File: pilot/pkg/xds/ads.go [hunk 2]
```diff
@@ -321,6 +315,10 @@ func (s *DiscoveryServer) Stream(stream DiscoveryStream) error {
...
+	if err = s.authorize(con, ids); err != nil {
```
Comment (howardjohn) [inline pilot/pkg/xds/ads.go:319]: This must happen before RegisterWorkload I feel like? Now its happening afterwards

**Review by howardjohn — COMMENTED** [review]
Abstracting the function SGTM. The new location is concerning.

File: pilot/pkg/xds/ads.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go:319]: will check

**Pull request merged**